### PR TITLE
PGI: add OpenMPI and pompi toolchain easyconfigs.

### PR DESCRIPTION
**Pull request by bartoldeman**
I hope to be able to uncomment the sanity_check_commands. For some strange reason, when they are run, `mpicc` does not find `pgcc` in the PATH; however after loading the module the `mpicc` command works just fine.

### CODE REVIEW

**Comment by bartoldeman** [comment]
(depends on hpcugent/easybuild-framework#1342 and #2879).

**Comment by boegel** [comment]
Jenkins: ok to test

**Comment by boegel** [comment]
@bartoldeman This PR is missing an `hwloc` easyconfig, fixed in https://github.com/bartoldeman/easybuild-easyconfigs/pull/2

File: easybuild/easyconfigs/o/OpenMPI/OpenMPI-1.10.2-PGI-16.3-GCC-4.9.3-2.25.eb [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+             ["lib/lib%s.%s" % (libfile, SHLIB_EXT) for libfile in libs] +
+             ["include/%s.h" % x for x in ["mpi-ext", "mpif-config", "mpif", "mpi", "mpi_portable_platform"]],
+    'dirs': ["include/openmpi/ompi/mpi/cxx"],
+}
+
+#sanity_check_commands = [
+#    ('mpicc --version | grep pgcc', ''),
+#    ('mpicxx --version | grep pgc++', ''),
+#    ('mpifort --version | grep pgfortran', ''),
+#]
```
Comment (boegel) [inline easybuild/easyconfigs/o/OpenMPI/OpenMPI-1.10.2-PGI-16.3-GCC-4.9.3-2.25.eb]: @bartoldeman we'll need to sort this out too...

**Comment by boegel** [comment]
@bartoldeman I traced these OpenMPI sanity check commands failing for you but not for me down to a bug in the EasyBuild framework, see https://github.com/hpcugent/easybuild-framework/pull/1754

The problem was that the `module load` being done at the start of the sanity check was not being done with the correct (short) module name, which may causes problems when using a hierarchical module naming scheme. The problem remained hidden up until now because it's only a real issue if: the toolchain being used is a non-bundle but an actual software package (like `PGI` is) *and* if there are sanity check commands listed that can fail if the module for the toolchain itself is not loaded (as is the case here).

Can you verify whether the fix in https://github.com/hpcugent/easybuild-framework/pull/1754 fixes the issue, i.e. whether you can install OpenMPI using the PGI toolchain with having these sanity check commands active using your hierarchical module naming scheme `ToolchainMNS`?

**Comment by bartoldeman** [comment]
It works with PR hpcugent/easybuild-framework#1754 so I uncommented those lines now.

**Comment by boegel** [comment]
@bartoldeman great, thanks for the confirmation!

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 3 out of 3 (3 easyconfigs in this PR)
node2159.delcatty.os - Linux SL 6.7, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/7b08be48f1ae9244b241feac5639f270 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 3 out of 3 (3 easyconfigs in this PR)
node2565.golett.os - Linux centos linux 7.2.1511, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/a024f9289ae5b56f397c4e9cd2696d0b for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 3 out of 3 (3 easyconfigs in this PR)
node2159.delcatty.os - Linux SL 6.7, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/7b7065bfcafbbe7f29b0cdb6bd6b0836 for a full test report.

**Comment by boegel** [comment]
Jenkins: test this please

**Comment by boegel** [comment]
Good to go now that https://github.com/hpcugent/easybuild-framework/pull/1342 is merged, thanks @bartoldeman!

**Pull request merged**